src/uproot/source/xrootd.py

```diff
@@ -111,7 +111,7 @@ def _xrd_error(self, status):
 
         # https://github.com/xrootd/xrootd/blob/8e91462e76ab969720b40fc324714b84e0b4bd42/src/XrdCl/XrdClStatus.hh#L47-L103
         # https://github.com/xrootd/xrootd/blob/250eced4d3787c2ac5be2c8c922134153bbf7f08/src/XrdCl/XrdClStatus.cc#L34-L74
-        if status.code == 101 or status.code == 304 or status.code == 400:
+        if status.code in {101, 304, 400}:
```

Comment:
Elsewhere, we have `if ___ in <tuple>`, so why should this one be a set?

```suggestion
        if status.code in (101, 304, 400):
```

You convinced me at one point that Python is smart enough to avoid the performance penalty that would come from constructing the tuple of constants just to see if some value is in it. That probably only applies to tuples of constants, not mutable types like a set (though it could, if a static code analyzer sees that this set doesn't escape the expression). Also, 3 is a small enough number of items that the hashmap-lookup might even be slower than iterating over the options.

But, of course, the main reason is for consistency. All of the other "is this a member of?" cases are tuples, not sets.

Reply:
This actually is stored as a frozen set in the byte code since it's inline :) But there's not much difference for a small one like this. To me, "in" is natural on a set, and bolted on for lists/tuples. But there are some things that can't go into sets. I'm not sure what the performance profile is, so no problem here, I just had to pick one and happy to pick the other.